client/src/components/productList/OrderItem/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { FilterContext } from '~/pages/ProductList';
+import { UnderlineBaeminColorSVG } from '~/assets';
+import { setOrder } from '~/stores/productListModule';
+
+import { TitleContainer, Title } from './index.style';
+
+interface Props {
+  content: string;
+  order: string;
+}
+
+const OrderItem: React.FC<Props> = ({ order, content }) => {
+  const { dispatch, ...currentState } = useContext(FilterContext);
+
+  const [isSelected, setIsSelected] = useState<boolean>(false);
+
+  const handleImgClick = () => {
+    if (order === 'recent') {
+      dispatch(setOrder('recent'));
+    }
+
+    if (order === 'high-price') {
+      dispatch(setOrder('high-price'));
+    }
+
+    if (order === 'low-price') {
+      dispatch(setOrder('low-price'));
+    }
+  };
+
+  useEffect(() => {
+    if (order === currentState.state.order) {
+      setIsSelected(true);
+    } else {
+      setIsSelected(false);
+    }
+  });
```

Comment:
조건 분기시 boolean 을 반환한다면 다음과 같이 수정할 수 있어 보입니다~

```suggestion
  useEffect(() => setIsSelected(order === currentState.state.order));
```

```suggestion
  useEffect(() => {
    const isSelectedOrder = order === currentState.state.order;
    setIsSelected(isSelectedOrder);
  });
```

Reply:
아 이부분은 타입 때문에 ㅠㅠㅠㅠㅠㅠㅠ 타입이 low-price, high-price, recent로 제한되어 있어서 string은 안된다고 하더라구요. type 쪽에서 수정을 해야할까요?